src/main/starlark/builtins_bzl/common/python/py_binary_bazel.bzl

```diff
@@ -44,5 +45,6 @@ def _py_binary_impl(ctx):
 py_binary = create_executable_rule(
     implementation = _py_binary_impl,
     attrs = AGNOSTIC_BINARY_ATTRS | _PY_TEST_ATTRS,
+    provides = [PyInfo],
```

Comment:
You'll also want this for py_test. Instead of setting it here, add the logic to create_executable_rule().

Reply:
Good! Yes, as was discussed here: https://github.com/bazelbuild/bazel/issues/19609#issuecomment-1759719928
I think it is very good for aspects to be able to run on py_tests as well, my reproduction does not have one unfortunately so I missed it. Good point to do it in the factory directly.

Reply:
I don't see any modification to the create_executable_rule function...?